base/math.jl

```diff
@@ -1308,14 +1326,37 @@ end
     return T(exp2(log2(abs(widen(x))) * y))
 end
 
-# compensated power by squaring
 @constprop :aggressive @inline function ^(x::Float64, n::Integer)
+    T = Int64
+    m = n >= typemax(T) ? typemax(T) : n <= typemin(T) ? typemin(T) : n
+    x^(m%Int64)
+end
```

Comment:
```suggestion
return x^clamp(n, Int64)
```

Reply:
`clamp` was detected buggy when I tried to use it here (#54038), therefore this work-around.

Reply:
I agree, that PR should be merged before this one.